encoding/x/protobuf/inbound.go

```diff
@@ -78,9 +81,16 @@ func (u *unaryHandler) Handle(ctx context.Context, transportRequest *transport.R
 		}
 		responseData = protoBuffer.Bytes()
 	}
+	if isRawResponse(transportRequest.Headers) {
+		responseWriter.AddHeaders(getRawResponseHeaders())
```

Comment:
See my comment on https://github.com/yarpc/yarpc-go/issues/905 and the linked commit - this is the only way to accomplish this two-way communication with the protobuf encoding, it is not possible to do this with the `Context` given the current yarpc API without adding what amounts to a two-way dependency.